src/main/java/org/folio/des/repository/criteria/CQL2JPACriteria.java

```diff
@@ -90,12 +90,24 @@ private CriteriaQuery<E> toCriteria(CQLNode node)
 
   private void processSort(CQLSortNode node) throws CQLFeatureUnsupportedException {
     List<Order> orders = new ArrayList<>();
+
     for (ModifierSet sortIndex : node.getSortIndexes()) {
       final CqlModifiers modifiers = new CqlModifiers(sortIndex);
-      orders.add(
-          CqlSort.DESCENDING.equals(modifiers.getCqlSort())
-              ? builder.desc(root.get(sortIndex.getBase()))
-              : builder.asc(root.get(sortIndex.getBase())));
+      String jsonPath = sortIndex.getBase();
+      int fieldNamesSize = JsonbNodeConverter.getFieldNames(jsonPath).size();
+
+      if (fieldNamesSize > 1){ // this case is for inner jsonb fields
+        Expression<String> innerField = JsonbNodeConverter.convertToExpression(root, jsonPath, builder);
+        orders.add(CqlSort.DESCENDING.equals(modifiers.getCqlSort())
```

Comment:
orders.add(CqlSort.DESCENDING.equals(modifiers.getCqlSort())
          ? builder.desc(innerField)
          : builder.asc(innerField));
I suggest to move to a separate method to avoid duplication

Reply:
Yeap, I extracted method for it